src/core/server/http/http_config.ts

```diff
@@ -87,11 +87,6 @@ const configSchema = schema.object(
     host: schema.string({
       defaultValue: 'localhost',
       hostname: true,
-      validate(value) {
-        if (value === '0') {
-          return 'value 0 is not a valid hostname (use "0.0.0.0" to bind to all interfaces)';
-        }
-      },
```

Comment:
`0` is now (correctly) considered as an invalid hostname by `joi`, so this custom validation is no longer required (and not invoked anyway if the value is `0`).

We'll have to use a `oneOf` here when backporting to `7.x` to allow `0`, as the branch is still supposed to support this value.